test/e2e/end_user_test.go

```diff
@@ -59,63 +62,32 @@ var _ = Describe("Openshift on Azure end user e2e tests [EndUser]", func() {
 	})
 
-	It("should deploy a template and check the visit counter increments", func() {
-		const TPL = "cakephp-mysql-example"
-		var regex = regexp.MustCompile(`id="count-value">(\d+)<`)
-
-		// Create the template
-		By(fmt.Sprintf("creating the template (%v)", time.Now()))
-		template, err := c.tc.Templates("openshift").Get(
-			TPL, metav1.GetOptions{})
-		Expect(err).NotTo(HaveOccurred())
-
-		// Instantiate the template
-		_, err = c.tc.TemplateInstances(c.namespace).Create(
-			&templatev1.TemplateInstance{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: c.namespace,
-				},
-				Spec: templatev1.TemplateInstanceSpec{
-					Template: *template,
-				},
-			})
-		Expect(err).NotTo(HaveOccurred())
-
-		// Poll for the deployment status
-		By(fmt.Sprintf("waiting for the template instance to turn ready (%v)", time.Now()))
-		err = wait.PollImmediate(2*time.Second, 5*time.Minute, c.templateInstanceIsReady)
+	It("should deploy a template and ensure a given text is in the contents", func() {
+		tpl := "nginx-example"
+		By(fmt.Sprintf("instantiating the template and getting the route (%v)", time.Now()))
+		// instantiate the template
+		err := c.instantiateTemplate(tpl)
 		Expect(err).NotTo(HaveOccurred())
-
-		// Pull the route ingress from the namespace
-		By(fmt.Sprintf("getting the route (%v)", time.Now()))
-		route, err := c.rc.Routes(c.namespace).Get(TPL, metav1.GetOptions{})
+		route, err := c.rc.Routes(c.namespace).Get(tpl, metav1.GetOptions{})
 		Expect(err).NotTo(HaveOccurred())
+		// make sure only 1 ingress point is returned
 		Expect(len(route.Status.Ingress)).To(Equal(1))
 		host := route.Status.Ingress[0].Host
 		url := fmt.Sprintf("http://%s", host)
 
-		prevCounter := 0
-		currCounter := 0
+		// Curl the endpoint and search for a string
 		httpc := &http.Client{
 			Timeout: 10 * time.Second,
 		}
-		By(fmt.Sprintf("hitting the route 3 times to check the counter (%v)", time.Now()))
-		// Hit the ingress 3 times, each time the hit counter should increment
-		for i := 0; i < 3; i++ {
-			resp, err := httpc.Get(url)
-			Expect(err).NotTo(HaveOccurred())
-			defer resp.Body.Close()
-			Expect(resp.StatusCode).Should(Equal(200))
-			contents, err := ioutil.ReadAll(resp.Body)
-			Expect(err).NotTo(HaveOccurred())
-			matches := regex.FindStringSubmatch(string(contents))
-			Expect(matches).NotTo(BeNil())
-			currCounter, err = strconv.Atoi(matches[1])
-			Expect(err).NotTo(HaveOccurred())
-			Expect(currCounter).Should(BeNumerically(">", prevCounter))
-			prevCounter = currCounter
-			time.Sleep(time.Second)
-		}
+		By(fmt.Sprintf("hitting the route and checking the contents (%v)", time.Now()))
+		resp, err := httpc.Get(url)
+		Expect(err).NotTo(HaveOccurred())
+		defer resp.Body.Close()
+		Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+		contents, err := ioutil.ReadAll(resp.Body)
+		Expect(err).NotTo(HaveOccurred())
+		found := strings.Contains(string(contents), "Welcome to your static nginx application on OpenShift")
```

Comment:
nit: or `Expect(contents).Should(ContainSubstring("Welcome to your static nginx application on OpenShift"))`